test/tools/yulInterpreter/EVMInstructionInterpreter.cpp

```diff
@@ -444,6 +444,26 @@ u256 EVMInstructionInterpreter::eval(
 	return 0;
 }
 
+u256 EVMInstructionInterpreter::evalBuiltin(YulString _name, const std::vector<u256>& _arguments)
+{
+	if (_name == "datasize"_yulstring)
+		return u256(keccak256(h256(_arguments.at(0)))) & 0xfff;
+	else if (_name == "dataoffset"_yulstring)
+		return u256(keccak256(h256(_arguments.at(0) + 2))) & 0xfff;
+	else if (_name == "datacopy"_yulstring)
+	{
+		// This is identical to codecopy.
+		if (logMemoryWrite(_arguments.at(0), _arguments.at(2)))
+			copyZeroExtended(
+				m_state.memory, m_state.code,
```

Comment:
I know that it's a little waste of lines, but shouldn't they all have a newline?

Reply:
oh yes of course!